src/main.rs

```diff
@@ -6,31 +6,89 @@ use skia_safe::Surface;
 use skia_safe::{Color, Data, IRect, Image, Point, RRect, Rect};
 
 use std::io::{self, Read, Write};
+use std::fs::{self, File};
+
+use clap::Parser;
 
 fn new_canvas(width: i32, height: i32) -> Surface {
     let mut surface = Surface::new_raster_n32_premul((width, height)).expect("no surface!");
     surface.canvas().clear(Color::TRANSPARENT);
     surface
 }
 
+struct ShadowColor {
+    r: u8,
+    g: u8,
+    b: u8,
+    a: u8,
+}
+
+impl From<String> for ShadowColor {
+    fn from(value: String) -> Self {
+        let r: u8 = u8::from_str_radix(&value[2..4],  16).unwrap_or(0);
+        let g: u8 = u8::from_str_radix(&value[4..6],  16).unwrap_or(0);
+        let b: u8 = u8::from_str_radix(&value[6..8],  16).unwrap_or(0);
+        let a: u8 = u8::from_str_radix(&value[8..10], 16).unwrap_or(0);
+
+        Self { r, g, b, a }
+    }
+}
+
+impl Into<Color> for ShadowColor {
+    fn into(self) -> Color {
+        Color::from_argb(self.a, self.r, self.g, self.b)
+    }
+}
+
+#[derive(Parser, Debug)]
+#[command(name = "shadower")]
+#[command(about = "Adds shadow, padding and rounded corners to images", long_about = None)]
+struct Args {
+    #[arg(short = 'r', long, default_value_t = 12.0, help = "border radius")]
+    radius: f32,
+    #[arg(short = 'p', long, default_value_t = 1./6., help = "padding = max((img.width * padding_ratio), (img.height * padding.ratio))")]
```

Comment:
these default values render as 0.16666667 in the help which doesn't really look good, otherwise LGTM

Reply:
found there's a `default_value` that takes in a string and parses it, thought it would act as the displayed one while `default_value_t` would be the actual one, turns out `default_value` overwrites `_t`